src/test/java/org/isf/medicalstock/test/Tests.java

```diff
@@ -1194,6 +1196,15 @@ public void testMovementGetOrigin() throws Exception {
 		assertThat(movement).isNotNull();
 		assertThat(movement.getSupplier()).isEqualTo(movement.getOrigin());
 	}
+	
+	@Test
+	public void testDeleteLastMovement() throws Exception {
+		int code = setupTestMovement(false);
+		Optional<Movement> movement = movementIoOperationRepository.findById(code); 
+		movBrowserManager.deleteLastMovement(movement.get());
```

Comment:
The previous statement returns an `Optional` which means it might be there or not there; yet you don't test for it and assume it is OK.   Please be more careful in your coding;

Yes I would agree that because it was just set up it should still be there but there is no guarantee that something unexpected happened.